### DISCUSSION TITLE
create image w/ multiple captions without an input image

### DISCUSSION THREAD

**Discussion by andykais** [ImageMagick/ImageMagick#3086] [Help] [answered]
I want to create transparent images containing text, that is no larger than the text it contains. [label](https://legacy.imagemagick.org/Usage/text/#label) are the right tool for the job, however I want to outline my text. The simple `-stroke`, `-strokewidth` flags are not desirable since they inset the stroke. Instead I want to layer two captions, one with a strokewidth behind, and one without a strokewidth in front. Similar to some of the [annotate examples](https://legacy.imagemagick.org/Usage/fonts/). I am able to achieve what I want with an input image:
```bash
convert logo: -background none -pointsize 30 -gravity Center -strokewidth 4 -stroke black -pointsize 30 'label:My Text Here' -compose over -composite -fill white -stroke none 'label:My Text Here' -compose over -composite out.png              
```
![out](https://user-images.githubusercontent.com/6627147/103843505-f064f800-5065-11eb-80ee-6121abd12088.png)

This is of course not the solution I want, since the output image is not transparent and is not the size of the label text.

Some older digging I did on this topic without much luck:
- https://legacy.imagemagick.org/discourse-server/viewtopic.php?t=16933
- https://legacy.imagemagick.org/discourse-server/viewtopic.php?t=15310

**Comment by snibgo** [comment, marked as answer]
Your command has three images: "logo:", then black text composited over that, then white text composited over that.

That is what you want, but without the logo, so you simply have white text composited over black text. Is that correct?
```
convert -background none -pointsize 30 -gravity Center -strokewidth 4 -stroke black -pointsize 30 'label:My Text Here' -fill white -stroke none 'label:My Text Here' -compose over -composite -trim +repage out.png  
```
I have added "-trim +repage" at the end to remove the small redundant space.

If you have IM v7, I suggest using "magick" instead of "convert".

**Reply by andykais** [reply]
Awesome, thanks. This does the trick!
